portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,35 +14,97 @@
 
 package com.google.sps.servlets;
 
+import java.util.Map;
+import java.util.HashMap;
 import com.google.gson.Gson;
-import com.google.common.collect.ImmutableList;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns inputted fun facts from users and stores previously inputted facts. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private static final ImmutableList<String> FRUIT_FACTS =
-    ImmutableList.of(
-      "A strawberry is not an actual berry", 
-      "Kiwis contain more Vitamin C than oranges", 
-      "The pineapple is actually a berry");
+  private static final String FIRST_NAME_KEY = "first-name";
+  private static final String LAST_NAME_KEY = "last-name";
+  private static final String FUN_FACT_KEY = "user-fun-fact";
+  private static final String NO_NAME = "Anonymous";
+  private Map<String, String> oldFacts = new HashMap<String, String>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String fruitFactsJson = convertToJsonUsingGson(FRUIT_FACTS);
+    String oldFactsJson = convertToJsonUsingGson(oldFacts);
 
     // Send the JSON as the response
     response.setContentType("application/json;");
-    response.getWriter().println(fruitFactsJson);
+    response.getWriter().println(oldFactsJson);
   }
 
-  private String convertToJsonUsingGson(ImmutableList<String> fruitFacts) {
+  private static String convertToJsonUsingGson(Map<String, String> oldFacts) {
     Gson gson = new Gson();
-    String json = gson.toJson(fruitFacts);
+    String json = gson.toJson(oldFacts);
     return json;
   }
-}
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String userFirstName = getParameter(request, /*content=*/ FIRST_NAME_KEY, /*defaultValue=*/ "");
+    String userLastName = getParameter(request, /*content=*/ LAST_NAME_KEY, /*defaultValue=*/ "");
+    String userFunFact = getParameter(request, /*content=*/ FUN_FACT_KEY, /*defaultValue=*/ "");
```

Comment:
Since you're using the name as a key to the map, it might be good to eventually clean the names by removing any initial or trailing white space so `"Chris"` is the same as `" Chris"` and `"Chris "`.

And do you care that `"Chris"` will be a different person from `"chris"` ?

Reply:
Yeah, I think that's definitely something I should consider. I think it would be best if I switch over to using unique usernames so I will do that in the next commit! Thank you for bringing this up!